cmd/memtierd/main.go

```diff
@@ -74,6 +74,55 @@ func loadConfigFile(filename string) (memtier.Policy, []memtier.Routine) {
 	return policy, routines
 }
 
+func processLoggerFlag(optLog string) {
+	switch optLog {
+	case "", "stderr":
+		memtier.SetLogger(log.New(os.Stderr, "", 0))
+	case "-", "stdout":
+		memtier.SetLogger(log.New(os.Stdout, "", 0))
+	default:
+		logFile, err := os.OpenFile(optLog, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
+		if err != nil {
+			exit("failed to open log file %q: %v", optLog, err)
+		}
+		memtier.SetLogger(log.New(logFile, "", 0))
+	}
+}
+
+func processOptCommandFileFlag(optCommandFile string, prompt *memtier.Prompt) {
+	if optCommandFile != "" {
+		commandFile, err := os.Open(optCommandFile)
+		if err != nil {
+			exit("error in opening command file %q: %v", optCommandFile, err)
+		}
+		prompt.SetInput(bufio.NewReader(commandFile))
+		memtier.Log().Debugf("executing commands from file %q", optCommandFile)
+		prompt.Interact()
+		commandFile.Close()
+	}
+}
+
+func commandPreCheck(optConfig, optCommandString, optCommandFile *string, optPrompt *bool) bool {
+	if *optConfig == "" && *optCommandString == "" && *optCommandFile == "" && !*optPrompt {
+		exit("required at least one of: -config CONFIGFILE, -c COMMANDS, -f COMMANDFILE, or -prompt")
+	}
+	return true
+
+}
+
+func setupRoutine(routines []memtier.Routine, policy memtier.Policy) {
```

Comment:
Most importantly this function starts routines (many of them). Could we rename it `startRoutines()`?